File: src/main/java/gargoyle/ct/task/helper/impl/CTTimeHelperImpl.java
package gargoyle.ct.task.helper.impl;

import gargoyle.ct.task.helper.CTTimeHelper;
import gargoyle.ct.util.CTTimeUtil;

public class CTTimeHelperImpl implements CTTimeHelper {
    private long delta;
    private long fakeTime;

    @Override
    public long currentTimeMillis() {
        return CTTimeUtil.currentTimeMillis() + delta;
    }

    @Override
    public long getFakeTime() {
        return fakeTime;
    }

    @Override
    public void setFakeTime(long fakeTime) {
        this.fakeTime = fakeTime;
        delta = fakeTime - CTTimeUtil.currentTimeMillis();
    }
}

File: src/main/kotlin/gargoyle/ct/task/helper/impl/CTTimeHelperImpl.kt
package gargoyle.ct.task.helper.impl

import gargoyle.ct.task.helper.CTTimeHelper
import gargoyle.ct.util.CTTimeUtil

class CTTimeHelperImpl : CTTimeHelper {
    private var delta: Long = 0
    override var fakeTime: Long = 0
        set(fakeTime) {
            field = fakeTime
            delta = fakeTime - CTTimeUtil.currentTimeMillis()
        }

    override fun currentTimeMillis(): Long {
        return CTTimeUtil.currentTimeMillis() + delta
    }
}
